crates/bevy_render/src/color/mod.rs

```diff
@@ -1083,69 +1083,224 @@ impl Color {
             }
         }
     }
-}
 
-impl Default for Color {
-    fn default() -> Self {
-        Color::WHITE
+    /// New `Color` from `Vec4` in sRGB colorspace.
+    #[inline]
+    pub fn rgba_from_vec4(vec4: Vec4) -> Self {
+        Color::rgba(vec4.x, vec4.y, vec4.z, vec4.w)
     }
-}
 
-impl AddAssign<Color> for Color {
-    fn add_assign(&mut self, rhs: Color) {
-        match self {
+    /// New `Color` from `[f32; 4]` in sRGB colorspace.
+    #[inline]
+    pub fn rgba_from_array([r, g, b, a]: [f32; 4]) -> Self {
```

Comment:
Would it make sense for these methods to take two parameters, with the second being the target color space? Or are we trying to keep the function signature `fn(Vec3) -> Color`?